packages/apollo-server-caching/src/testsuite.ts

```diff
@@ -0,0 +1,56 @@
+import { KeyValueCache } from './KeyValueCache';
+
+/**
+ * runKeyValueCacheTests is a function you can call from a test that exercises
+ * the given KeyValueCache. It throws on failure. If you want to test
+ * expiration, then mock out `Date` and `setTimeout` (probably with
+ * `@sinonjs/fake-timers`) and pass a `tick` that can be called to advance the
+ * fake time. (If you don't pass `tick`, it won't test expiration.) Other than
+ * that, it has no dependencies and can work in any test system and shouldn't
+ * require any particular build configuration to use from jest. See the
+ * README.md for an example of how to use this with jest.
+ */
+export async function runKeyValueCacheTests(
```

Comment:
I think this is a fine solution for now, to avoid creating a separate testing package.  An alternative would have been to define a separate TypeScript project within the same package, which would have allowed the test to depend on Jest without those types affecting the production code. One benefit of that is that it would keep the ability to report individual test failures, as opposed to the first assertion throwing.

Reply:
My concern is that I do want this to be easily available to other implementations outside of the repo, and the vague (perhaps flawed) research I tried to do on how to make a Jest suite "exportable" didn't turn up much. I didn't want to put a bunch of effort into something that might end up not working in different TS/Jest contexts when it seemed reasonably straightforward to write something that would definitely work all the time.  Reporting individual test failures would be nice, but it's also like one page of code in a package that hasn't been a major focus of Apollo first-party development recently.